clients/src/main/java/org/apache/kafka/clients/producer/internals/ErrorLoggingCallback.java

```diff
@@ -17,40 +17,22 @@
 package org.apache.kafka.clients.producer.internals;
 
 import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.ProducerRecord;
 import org.apache.kafka.clients.producer.RecordMetadata;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.nio.charset.StandardCharsets;
-
 public class ErrorLoggingCallback implements Callback {
     private static final Logger log = LoggerFactory.getLogger(ErrorLoggingCallback.class);
-    private String topic;
-    private byte[] key;
-    private byte[] value;
-    private int valueLength;
-    private boolean logAsString;
-
-    public ErrorLoggingCallback(String topic, byte[] key, byte[] value, boolean logAsString) {
-        this.topic = topic;
-        this.key = key;
-
-        if (logAsString) {
-            this.value = value;
-        }
+    private ProducerRecord record;
 
-        this.valueLength = value == null ? -1 : value.length;
-        this.logAsString = logAsString;
+    public ErrorLoggingCallback(ProducerRecord record) {
+        this.record = record;
     }
 
     public void onCompletion(RecordMetadata metadata, Exception e) {
         if (e != null) {
-            String keyString = (key == null) ? "null" :
-                    logAsString ? new String(key, StandardCharsets.UTF_8) : key.length + " bytes";
-            String valueString = (valueLength == -1) ? "null" :
-                    logAsString ? new String(value, StandardCharsets.UTF_8) : valueLength + " bytes";
-            log.error("Error when sending message to topic {} with key: {}, value: {} with error:",
-                    topic, keyString, valueString, e);
+            log.error("Error when sending {}", record);
```

Comment:
Include the exception in the log so there is an indication of what went wrong? Looks like it was previously included.